main/Main.hs

```diff
@@ -2,180 +2,290 @@
 
 module Main where
 
-import           Protolude
-import           PostgREST.App
-import           PostgREST.Config                     (AppConfig (..),
-                                                       PgVersion (..),
-                                                       minimumPgVersion,
-                                                       prettyVersion,
-                                                       readOptions)
-import           PostgREST.Error                      (encodeError)
-import           PostgREST.OpenAPI                    (isMalformedProxyUri)
-import           PostgREST.DbStructure
-import           PostgREST.Types                      (DbStructure, Schema)
-
-import           Control.AutoUpdate
-import           Control.Retry
-import           Data.ByteString.Base64               (decode)
-import           Data.String                          (IsString (..))
-import           Data.Text                            (stripPrefix, pack, replace)
-import           Data.Text.Encoding                   (encodeUtf8, decodeUtf8)
-import           Data.Text.IO                         (hPutStrLn, readFile)
-import           Data.Time.Clock.POSIX                (getPOSIXTime)
-import qualified Hasql.Query                          as H
-import qualified Hasql.Session                        as H
-import qualified Hasql.Decoders                       as HD
-import qualified Hasql.Encoders                       as HE
-import qualified Hasql.Pool                           as P
-import           Network.Wai.Handler.Warp
-import           System.IO                            (BufferMode (..),
-                                                       hSetBuffering)
-import           Data.IORef
+import PostgREST.App (postgrest)
+import PostgREST.Config
+       (AppConfig(..), PgVersion(..), minimumPgVersion, prettyVersion,
+        readOptions)
+import PostgREST.DbStructure (getDbStructure)
+import PostgREST.Error (encodeError)
+import PostgREST.OpenAPI (isMalformedProxyUri)
+import PostgREST.Types (DbStructure, Schema)
+import Protolude
+
+import Control.AutoUpdate
+import Control.Retry
+import Data.ByteString.Base64 (decode)
+import Data.IORef (IORef, atomicWriteIORef, newIORef, readIORef)
+import Data.String (IsString(..))
+import Data.Text (pack, replace, stripPrefix)
+import Data.Text.Encoding (decodeUtf8, encodeUtf8)
+import Data.Text.IO (hPutStrLn, readFile)
+import Data.Time.Clock.POSIX (getPOSIXTime)
+import qualified Hasql.Decoders as HD
+import qualified Hasql.Encoders as HE
+import qualified Hasql.Pool as P
+import qualified Hasql.Query as H
+import qualified Hasql.Session as H
+import Network.Wai.Handler.Warp
+       (defaultSettings, runSettings, setHost, setPort, setServerName,
+        setTimeout)
+import System.IO (BufferMode(..), hSetBuffering)
```

Comment:
I agree about the old alignment being nice. The suggestions in [this article](https://hackernoon.com/4-steps-to-a-better-imports-list-in-haskell-43a3d868273c) about imports seem pretty good. Maybe you could review our import lists and get them all to follow those guidelines.